app/Http/Controllers/V2/Terrafund/TerrafundEditGeometryController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers\V2\Terrafund;
+
+use App\Http\Controllers\Controller;
+use App\Models\V2\PolygonGeometry;
+use App\Models\V2\Sites\SitePolygon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
+
+class TerrafundEditGeometryController extends Controller
+{
+    public function getSitePolygonData(string $uuid)
+    {
+        try {
+            $sitePolygon = SitePolygon::where('poly_id', $uuid)->first();
+
+            if (! $sitePolygon) {
+                return response()->json(['message' => 'No site polygons found for the given UUID.'], 404);
+            }
+
+            return response()->json(['site_polygon' => $sitePolygon]);
+        } catch (\Exception $e) {
+            return response()->json(['message' => $e->getMessage()], 500);
+        }
+    }
+
+    public function updateGeometry(string $uuid, Request $request)
+    {
+        $geometry = json_decode($request->input('geometry'));
+        $geom = DB::raw("ST_GeomFromGeoJSON('" . json_encode($geometry) . "')");
+        $polygonGeometry = PolygonGeometry::where('uuid', $uuid)->first();
+        if (! $polygonGeometry) {
+            return response()->json(['message' => 'No polygon geometry found for the given UUID.'], 404);
+        }
+        $polygonGeometry->geom = $geom;
+        $polygonGeometry->save();
+
+        return response()->json(['message' => 'Geometry updated successfully.', 'geometry' => $geometry, 'uuid' => $uuid]);
+    }
+
+    public function getPolygonGeojson(string $uuid)
+    {
+        // get the st_geojson from polygon_geometry
+        $polygonGeometry = PolygonGeometry::where('uuid', $uuid)->first();
+        if (! $polygonGeometry) {
+            return response()->json(['message' => 'No polygon geometry found for the given UUID.'], 404);
+        }
+        $geojson = DB::table('polygon_geometry')
+        ->select(DB::raw('ST_AsGeoJSON(geom) as geojson'))
+        ->where('uuid', '=', $uuid)
+        ->get();
+
+        $geojsonData = json_decode($geojson[0]->geojson, true);
```

Comment:
This can be a bit cheaper in the DB and a little easier to read:
```
$geometryQuery = PolygonGeometry::isUuid($uuid);
if (! $gemoetryQuery->exists()) {
  return response()->json(['message' => 'No polygon geometry found for the given UUID.'], 404);
}

$geojsonData = json_decode($geometryQuery->select(DB::raw('ST_AsGeoJSON(geom) as geojson'))->first()->geojson, true));
```

Reply:
improved code with the suggestion